### PR TITLE
Add new Westeros tutorial: Addon technologies

### PR DESCRIPTION
**Pull request by OFR-IIASA**
This PR adds a new tutorial for explaining how to add "addon" technologies.

## How to review

Run tutorial
Read through tutorials and see if the explanantions are understandable 

## PR checklist

- [x] Tests added.
- [x] Documentation added.
- [x] Release notes updated.

### CODE REVIEW

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 1]
```diff
@@ -0,0 +1,607 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Westeros Tutorial\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:7]: Use a title format consistent with other Westeros tutorials, and possibly numbering as you wish to bring this after a certain tutorial, like 3c.

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+    "\n",
+    "What this means, is that the additional technology options are explicitly modelled as separate technologies, classified as `addon` technologies, linked to the activity of the technology to which they serve as additional configuration options, the parent technology, through specifying a `addon_conversion` factor.  The activity of the `addon` technology can further be restricted to a minimum or maximum share of the activity of the parent technology."
+   ]
+  },
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:16]: Mentio "requirements" for running this tutorial, like running "baseline" or any other tutorial beforehand. (please refer to other tutorials to see examples).

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adding heat co-generation capacibility to the `coal_ppl`\n",
+    "This tutorial will extend the current reference-energy-system to include a demand for heat and the necessary technologies to meet this demand.  Heat will be generated via a `passout-turbine` which will be linked to the `coal_ppl` using the `addon` formulation.\n",
+    "\n",
+    "# *Add modified RES*"
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:24]: Please apply an intiutive heading numbers, size, and format that helps the user to navigate through the tutorial steps.

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 4]
```diff
@@ -0,0 +1,607 @@
...
+   "source": [
+    "mp = ixmp.Platform()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load the existing scenario '*baseline*' and clone to a new scenario '*addon_technology*' which we will extended to include heat cogeneration using the `addon` feature"
+   ]
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:56]: Please convert long titles/headings to short and concise ones, and bring the text below the heading.

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 5]
```diff
@@ -0,0 +1,607 @@
...
+    "\n",
+    "base_output = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'mode': 'standard',\n",
+    "    'node_dest': country,\n",
+    "    'time': 'year',\n",
+    "    'time_dest': 'year', \n",
+    "    'unit': '%',\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:123]: Please make sure the unit is correct (% seems incorrect here).

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 6]
```diff
@@ -0,0 +1,607 @@
...
+    "}\n",
+    "base_capacity_factor = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'time': 'year',\n",
+    "    'unit': '%',\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:130]: Please make sure the unit is correct (% seems should be replaced with '-').

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 7]
```diff
@@ -0,0 +1,607 @@
...
+    "}\n",
+    "base_technical_lifetime = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': model_horizon,\n",
+    "    'unit': 'y',\n",
+    "}\n",
+    "base_inv_cost = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': model_horizon,\n",
+    "    'unit': 'USD/GWa',\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:140]: Please make sure the unit is correct (USD/GWa seems should be USD/kW).

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 8]
```diff
@@ -0,0 +1,607 @@
...
+    "    'year_act': act_years,\n",
+    "    'unit': 'USD/GWa',\n",
+    "}\n",
+    "base_var_cost = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'mode': 'standard',\n",
+    "    'time': 'year',\n",
+    "    'unit': 'USD/GWa',\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:154]: Please make sure the unit is correct (USD/GWa seems should be USD/kWa).

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 9]
```diff
@@ -0,0 +1,607 @@
...
+    "base_inv_cost = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': model_horizon,\n",
+    "    'unit': 'USD/GWa',\n",
+    "}\n",
+    "base_fix_cost = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': vintage_years,\n",
+    "    'year_act': act_years,\n",
+    "    'unit': 'USD/GWa',\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:146]: Please make sure the unit is correct (USD/GWa seems should be USD/kW).

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 10]
```diff
@@ -0,0 +1,607 @@
...
+   "source": [
+    "tec = 'po_turbine'\n",
+    "scen.add_set('technology', tec)\n",
+    "\n",
+    "po_out = make_df(base_output, technology=tec, commodity='heat', \n",
+    "                   level='secondary', value=1.0)\n",
+    "scen.add_par('output', po_out)\n",
+    "\n",
+    "po_in = make_df(base_input, technology=tec, commodity='electricity',  \n",
+    "                  level='secondary', value=0.2, unit='%')\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:283]: Please make sure there is explanation for using this number 0.2.

File: tutorial/westeros/westeros_addon_technologies.ipynb [hunk 11]
```diff
@@ -0,0 +1,607 @@
...
+    "#                                  'year_act', 'mode', 'time',\n",
+    "#                                  'type_addon', 'value', 'unit']\n",
+    "df = pd.DataFrame({'node': country,\n",
+    "                   'technology': tec,\n",
+    "                   'year_vtg': vintage_years,\n",
+    "                   'year_act': act_years,\n",
+    "                   'mode': 'standard',\n",
+    "                   'time': 'year',\n",
+    "                   'type_addon': type_addon,\n",
+    "                   'value': 5,\n",
```
Comment (behnam-zakeri) [inline tutorial/westeros/westeros_addon_technologies.ipynb:368]: Please make sure this value 5 is explained. is this related to 0.2 input electricity of po_turbine?

**Review by behnam-zakeri — CHANGES_REQUESTED** [review]
@OFR-IIASA, thank you so much for this useful tutorial. Please consider the following changes/improvements:
1. Using a title format consistent with other Westeros tutorials, and possibly numbering as you wish to bring this after a certain tutorial, like 3c.
2. Mentioning "requirements" for running this tutorial, like running "baseline" or any other tutorial beforehand. (please refer to other tutorials to see examples). 
3. Possibly apply an intiutive heading numbers, format and size, that helps moving through the tutorial.
4. Reviewing and updating all the units of parameters.
I also added some minor comments related to the numbers being calculated.

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 1 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:7]: done.

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 2 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:16]: done

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 3 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:24]: This in fact is not a heading, but it means that I still need to add a modified version of the RES - i just dont have the source document used to generate the Westeros RES.

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 5 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:123]: We can change this to something else, but it is actually an efficiency. It means that for every 5 units of activity from the coal_ppl, 1 unit of activity from the po_trubine can be generated.

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 6 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:130]: again, I am not sure about this, as this is a share. it indicates that a maximum of 15% of the activity of coal_ppl can be used to generate heat.

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 9 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:146]: done

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 7 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:140]: done

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 8 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:154]: done

File: tutorial/westeros/westeros_addon_technologies.ipynb (hunk 10 above)
Comment (OFR-IIASA) [reply tutorial/westeros/westeros_addon_technologies.ipynb:283]: it was actually explained above, but i detailed this some more as it wasnt clear enough.

**Comment by behnam-zakeri** [comment]
@OFR-IIASA , thanks for this. I suggested some typo corrections. Also, it will be useful to see the technology name in RES presented as "heating", more closely related to the technology name in the code (currently `dh_house`). I suggest changing the `dh_house` to `hs_house` and the label in RES to "heating system". What do you think?

**Comment by behnam-zakeri** [comment]
@Thanks Oliver for the corrections. @khaeru this PR is ready to be merged. I checked it for the content and ran it. It also passed all the checks. If you don't have any concerns, I can merge it.

**Review by behnam-zakeri — APPROVED** [review]

**Comment by khaeru** [comment]
> @khaeru this PR is ready to be merged. I checked it for the content and ran it. It also passed all the checks. If you don't have any concerns, I can merge it.

Thanks for the review, tag, and info! The conflict is due to changes (simplifications) I made in #463 (sorry!) so it's on me to clean up. I'll then merge.

**Comment by khaeru** [comment]
Many thanks @OFR-IIASA for the new tutorial! 🎉

**Pull request merged**